File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-0.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        n = len(arr)
        m = -1
        for i in range(n - 1, -1, -1):
    
            tmp = max(m, arr[i])
            arr[i] = m
            m = tmp
        return arr

File: Data Structures & Algorithms/append-characters-to-string-to-make-subsequence/submission-0.cpp
class Solution {
public:
    int appendCharacters(string s, string t) {
        int p1 = 0, p2 = 0;
        int n1 = s.size(), n2 = t.size();
        while(p1 < n1 && p2 < n2){
            if (s[p1] == t[p2]) p2++;
            p1++;
        }
        return n2 - p2;
    }
};